bibliopixel/remote/control.py

```diff
@@ -14,89 +13,95 @@
     'valid': True
 }
 
-DEFAULT_SERVER_CONFIG = {
-    'title': 'BiblioPixel Remote',
-    'bgcolor': '#000000',
-    'font_color': '#ffffff',
-    'external_access': False,
-    'port': 5000
+
+OFF_ANIM = {
+    'animation': 'bibliopixel.animation.off.OffAnim',
+    'run': {'threaded': True}
 }
 
+
 BAD_DEFAULT_ERROR = """\
 `{}` is not a valid default!
 It must be one of the configured animation names."""
 
 
-class RemoteControl:
-    def __init__(self, config, animations):
-        # These start with the defaults
-        self.config = dict(DEFAULT_SERVER_CONFIG, **config)
+class RemoteControl(collection.Collection):
+    def __init__(self, layout, animations,
+                 external_access=False, port=5000,
+                 title='BiblioPixel Remote', bgcolor='black',
+                 font_color='white', default=None):
+
+        self.internal_delay = 0  # never wait
+
+        # normalize name and display first
+        for anim in animations:
+            anim['animation']['data'] = dict(DEFAULT_ANIM_CONFIG, **anim['animation'].get('data', {}))
+            if not anim['animation'].get('name'):
+                raise ValueError('All animations require the `name` parameter: {}'.format(anim['animation']))
```

Comment:
oh man... I did not know about that enumerate trick! I always want to get the object and it's position. Very cool.
And yeah... agreed. My other though was to do an inspection and name the animation whatever the class name was...